### ISSUE TITLE
Gated styles: Ensure all prompts to remove premium styles automatically works

### ISSUE THREAD

**Issue by annezazu** [Automattic/wp-calypso#97730]
Right now, if you go through onboarding and select a theme with paid style variations, you're prompted to upgrade or remove premium styles in three places: Launchpad, Front end, Site Editor. The problem is only the "remove premium styles" option in the site editor actually removes premium styles for the user, resolving the problem. At this stage, particularly during onboarding and before they've really made many style changes, we should automate this as much as possible. 

I propose we automatically remove premium styles in one click at each stage, matching what's done in the site editor but taking it a step further where someone doesn't need to click "save" and it's just handled. 

**Launchpad**

It's a dead end to show this here as the user can't even rectify it in the site editor as they haven't even been introduced to it yet! We should either automatically set to the default palette to remove premium styles or send them back to select a different theme (I say the former):

https://github.com/user-attachments/assets/a1e842a3-8675-40c6-bc72-cc4115ed0bd0

**Front end**

https://github.com/user-attachments/assets/60f47903-14f8-4a55-9fc9-390b252204e6

**Site Editor**

We still require folks to hit "save" here and I wonder if there's a way we can automatically save for them when other styling changes haven't been made. 

https://github.com/user-attachments/assets/70403f49-d82a-4674-8a1e-1a70741b6b02

**Comment by annezazu** [comment]
Related! https://github.com/Automattic/wp-calypso/issues/95578

**Comment by ramonjd** [comment]
I think we should be able to come up with a hook to do it all: reset global styles and save. 

Agree: ideally, the user shouldn't be brought to the editor in the first instance at all.

**Comment by ramonjd** [comment]
> I think we should be able to come up with a hook to do it all: reset global styles and save.

It would be a useful Gutenberg hook perhaps 😄

**Comment by ramonjd** [comment]
I was just looking at this today, and it's a bit tricky 🤔 

Here's the Jetpack feature code repo for reference: https://github.com/Automattic/jetpack/tree/9a96a302b7e6229a49578102f59cc7292ae49cf9/projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles

The reset links ([this one for example](https://github.com/Automattic/jetpack/blob/9a96a302b7e6229a49578102f59cc7292ae49cf9/projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php#L540-L540)) that sit outside of the editor context do not have access to core packages, or the editor store, and therefore any selectors/actions to:

1. fetch the current global styles entity ID for the installed theme
2. reset that entity

I think it might be possible by creating a custom vanilla JS hook and utilizing the [Global_Styles_Status_Rest_API](https://github.com/Automattic/jetpack/blob/9a96a302b7e6229a49578102f59cc7292ae49cf9/projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/api/class-global-styles-status-rest-api.php) class to create a new reset/save endpoint.

**Comment by ramonjd** [comment]
Starting testing over in https://github.com/Automattic/jetpack/pull/40818. Taking longer as I familiarize myself with the processes.

**Comment by richtabor** [comment]
We should do a bit of design thinking on the UX here as well cc @crisbusquets. 

This is odd having a toggle, with an upgrade and an external link, perhaps in the wrong order as well. 

Perhaps a ToggleGroup with "Premium" and "Free" options, that when toggled reveal the effects of those styles — i.e. selecting "Free" removes the styles from the site, so you can preview it.

Then below it, a call to action, either "Upgrade" or  "Remove Styles"

Here's a quick sketch: 

 ![Image](https://github.com/user-attachments/assets/d5dd9699-8f7d-47a9-9dc0-7cd66eca8c69)

![Image](https://github.com/user-attachments/assets/b6f2b4e4-c708-4023-a44e-9c153ac0c6f6)

**Comment by ramonjd** [comment]
I'm also wondering if remove styles is even required in the frontend? The CTA serves as a constant reminder that the styles don't appear on the site, and, if users do ever upgrade, the styles would appear as the user selected: they wouldn't have to go back and restore the revision.

**Comment by crisbusquets** [comment]
> We should do a bit of design thinking on the UX here as well cc [@crisbusquets](https://github.com/crisbusquets).

Thanks for the ping, @richtabor ! Yes, I totally agree that this _tooltip_ doesn't feel right. I like the `Site Edit` banner approach—perhaps we should aim for something like that instead of what's in the screenshot below, which requires an extra click to open the tooltip:

![Image](https://github.com/user-attachments/assets/8cb333c3-f76d-4ee2-942f-54e6403f3f5d)

Could we use the gray background as if it was a banner?

@nuriapenya this is related to what we've been discussing about creating an upsell pattern/language that we can reuse.

**Comment by richtabor** [comment]
> I like the Site Edit banner approach—perhaps

Agreed. We should consider a design where this banner notice is solely for the global styles upgrade (no other CTAs etc). It does not have to be a popover.

**Comment by annezazu** [comment]
Created https://github.com/Automattic/wp-calypso/issues/98039 to split out the lovely design discussion here! Very excited to see this getting some love and attention.

**Comment by ramonjd** [comment]
Update, while I'm off to work on other things:

The Jetpack PR for the logged-in popover during site preview is functionally ready: 

- https://github.com/Automattic/jetpack/pull/40818

It doesn't implement a radical design change as there are outstanding questions about whether, or how, to display the preview buttons. 

I think removing the need for the user to click on a WordPress doc to work out how to remove the global styles themselves is the primary UX improvement we can ship here in the meantime. 

On the Calypso side, a similar treatment would be required here to be changed here: https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/plan/index.tsx#L32
